test/tools/ossfuzz/proto_to_yul.cpp

```diff
@@ -0,0 +1,234 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/tools/ossfuzz/proto_to_yul.h>
+#include <test/tools/ossfuzz/yul_proto.pb.h>
+
+#include <ostream>
+#include <sstream>
+
+namespace yul_fuzzer
+{
+// Forward decls.
+std::ostream& operator<<(std::ostream& _os, const BinaryOp& _x);
+
+std::ostream& operator<<(std::ostream& _os, const StatementSeq& _x);
+
+// Proto to yul.
+// Constant is a signed 32-bit integer
+std::ostream& operator<<(std::ostream& _os, const Const& _x)
+{
+	return _os << "(" << _x.val() << ")";
+}
+
+std::ostream& operator<<(std::ostream& _os , const VarRef& _x)
+{
+	return _os  << "x_" << (static_cast<uint32_t>(_x.varnum()) % 10);
+}
+
+std::ostream& operator<<(std::ostream& _os , const Lvalue& _x)
+{
+	return _os  << _x.varref();
+}
+
+std::ostream& operator<<(std::ostream& _os , const Rvalue& _x)
+{
+	if (_x.has_varref())
+		return _os  << _x.varref();
+	if (_x.has_cons())
+		return _os  << _x.cons();
+	if (_x.has_binop())
+		return _os  << _x.binop();
+	return _os  << "1";
+}
+
+std::ostream& operator<<(std::ostream& _os , const BinaryOp& _x)
+{
+	switch (_x.op()) {
+		case BinaryOp::ADD:
+			_os << "add(" << _x.left() << "," << _x.right() << ")";
```

Comment:
This could be simplified to just prefixing the operator name inside the switch and then calling `left()` and `right()` only once at the end.

Reply:
done